### PR TITLE
`azurerm_sentinel_alert_rule_scheduled` - support for new values in `tactics` property

### LINKED ISSUE #18013: Value InhibitResponseFunction missing for azurerm_sentinel_alert_rule_scheduled tactics
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.2.6

### AzureRM Provider Version

3.16.0

### Affected Resource(s)/Data Source(s)

azurerm_sentinel_alert_rule_scheduled

### Terraform Configuration Files

```hcl
resource "azurerm_sentinel_alert_rule_scheduled" "ar" {
    name                       = "ar"
    display_name               = "Azure DDoS Protection Detected an attack"
    severity                   = "High"
    description                = "The Azure DDoS Protection Standard service has detected an active attack towards a public IP address."

    event_grouping {
        aggregation_method = "AlertPerResult"
    }

    incident_configuration {
        create_incident = true
        grouping {
            enabled                = true
            entity_matching_method = "AllEntities"
            lookback_duration      = "P1D"
        }
    }

    query_period      = "PT5M"
    query_frequency   = "PT5M"
    trigger_operator  = "GreaterThan"
    trigger_threshold = 0
    tactics           = [
        "InhibitResponseFunction",
    ]

    entity_mapping {
        entity_type = "IP"
        field_mapping {
            column_name = "PublicIPAddress"
            identifier  = "Address"
        }
    }
    entity_mapping {
        entity_type = "AzureResource"
        field_mapping {
            column_name = "_ResourceId"
            identifier  = "ResourceId"
        }
    }


    query = <<QUERY
AzureDiagnostics
| where Category == "DDoSProtectionNotifications"
    QUERY

    log_analytics_workspace_id = var.log_analytics_workspace_id
}
```

### Debug Output/Panic Output

```shell
Error: expected tactics.0 to be one of [Collection CommandAndControl CredentialAccess DefenseEvasion Discovery Execution Exfiltration Impact InitialAccess LateralMovement Persistence PrivilegeEscalation PreAttack], got InhibitResponseFunction
```

### Expected Behaviour

InhibitResponseFunction is an allowed value

### Actual Behaviour

InhibitResponseFunction fails

### Steps to Reproduce

Apply with tactics = ["InhibitResponseFunction"]

### Important Factoids

_No response_

### References

_No response_

### PR DESCRIPTION
**Pull request by ziyeqf**
upgrade `securityinsight` to `2022-01-01-preview` to fix #18013

<details>
<summary>Test</summary>

```
TF_ACC=1 go test -v ./internal/services/sentinel -run=TestAccSentinel -timeout=600m

=== RUN   TestAccSentinelAlertRuleDataSource_basic
=== PAUSE TestAccSentinelAlertRuleDataSource_basic
=== RUN   TestAccSentinelAlertRuleFusion_basic
=== PAUSE TestAccSentinelAlertRuleFusion_basic
=== RUN   TestAccSentinelAlertRuleFusion_complete
=== PAUSE TestAccSentinelAlertRuleFusion_complete
=== RUN   TestAccSentinelAlertRuleFusion_update
=== PAUSE TestAccSentinelAlertRuleFusion_update
=== RUN   TestAccSentinelAlertRuleFusion_requiresImport
=== PAUSE TestAccSentinelAlertRuleFusion_requiresImport
=== RUN   TestAccSentinelAlertRuleMLBehaviorAnalytics_basic
=== PAUSE TestAccSentinelAlertRuleMLBehaviorAnalytics_basic
=== RUN   TestAccSentinelAlertRuleMLBehaviorAnalytics_complete
=== PAUSE TestAccSentinelAlertRuleMLBehaviorAnalytics_complete
=== RUN   TestAccSentinelAlertRuleMLBehaviorAnalytics_update
=== PAUSE TestAccSentinelAlertRuleMLBehaviorAnalytics_update
=== RUN   TestAccSentinelAlertRuleMLBehaviorAnalytics_requiresImport
=== PAUSE TestAccSentinelAlertRuleMLBehaviorAnalytics_requiresImport
=== RUN   TestAccSentinelAlertRuleMsSecurityIncident_basic
=== PAUSE TestAccSentinelAlertRuleMsSecurityIncident_basic
=== RUN   TestAccSentinelAlertRuleMsSecurityIncident_complete
=== PAUSE TestAccSentinelAlertRuleMsSecurityIncident_complete
=== RUN   TestAccSentinelAlertRuleMsSecurityIncident_update
=== PAUSE TestAccSentinelAlertRuleMsSecurityIncident_update
=== RUN   TestAccSentinelAlertRuleMsSecurityIncident_requiresImport
=== PAUSE TestAccSentinelAlertRuleMsSecurityIncident_requiresImport
=== RUN   TestAccSentinelAlertRuleMsSecurityIncident_withAlertRuleTemplateGuid
=== PAUSE TestAccSentinelAlertRuleMsSecurityIncident_withAlertRuleTemplateGuid
=== RUN   TestAccSentinelAlertRuleMsSecurityIncident_withDisplayNameExcludeFilter
=== PAUSE TestAccSentinelAlertRuleMsSecurityIncident_withDisplayNameExcludeFilter
=== RUN   TestAccSentinelAlertRuleNrt_basic
=== PAUSE TestAccSentinelAlertRuleNrt_basic
=== RUN   TestAccSentinelAlertRuleNrt_complete
=== PAUSE TestAccSentinelAlertRuleNrt_complete
=== RUN   TestAccSentinelAlertRuleNrt_update
=== PAUSE TestAccSentinelAlertRuleNrt_update
=== RUN   TestAccSentinelAlertRuleNrt_requiresImport
=== PAUSE TestAccSentinelAlertRuleNrt_requiresImport
=== RUN   TestAccSentinelAlertRuleNrt_withAlertRuleTemplateGuid
=== PAUSE TestAccSentinelAlertRuleNrt_withAlertRuleTemplateGuid
=== RUN   TestAccSentinelAlertRuleScheduled_basic
=== PAUSE TestAccSentinelAlertRuleScheduled_basic
=== RUN   TestAccSentinelAlertRuleScheduled_complete
=== PAUSE TestAccSentinelAlertRuleScheduled_complete
=== RUN   TestAccSentinelAlertRuleScheduled_update
=== PAUSE TestAccSentinelAlertRuleScheduled_update
=== RUN   TestAccSentinelAlertRuleScheduled_requiresImport
=== PAUSE TestAccSentinelAlertRuleScheduled_requiresImport
=== RUN   TestAccSentinelAlertRuleScheduled_withAlertRuleTemplateGuid
=== PAUSE TestAccSentinelAlertRuleScheduled_withAlertRuleTemplateGuid
=== RUN   TestAccSentinelAlertRuleScheduled_updateEventGroupingSetting
=== PAUSE TestAccSentinelAlertRuleScheduled_updateEventGroupingSetting
=== RUN   TestAccSentinelAlertRuleTemplateDataSource_fusion
=== PAUSE TestAccSentinelAlertRuleTemplateDataSource_fusion
=== RUN   TestAccSentinelAlertRuleTemplateDataSource_securityIncident
=== PAUSE TestAccSentinelAlertRuleTemplateDataSource_securityIncident
=== RUN   TestAccSentinelAlertRuleTemplateDataSource_scheduled
=== PAUSE TestAccSentinelAlertRuleTemplateDataSource_scheduled
=== RUN   TestAccSentinelAlertRuleTemplateDataSource_nrt
=== PAUSE TestAccSentinelAlertRuleTemplateDataSource_nrt
=== RUN   TestAccSentinelAutomationRule_basic
=== PAUSE TestAccSentinelAutomationRule_basic
=== RUN   TestAccSentinelAutomationRule_complete
=== PAUSE TestAccSentinelAutomationRule_complete
=== RUN   TestAccSentinelAutomationRule_update
=== PAUSE TestAccSentinelAutomationRule_update
=== RUN   TestAccSentinelAutomationRule_requiresImport
=== PAUSE TestAccSentinelAutomationRule_requiresImport
=== CONT  TestAccSentinelAlertRuleDataSource_basic
=== CONT  TestAccSentinelAlertRuleNrt_requiresImport
--- PASS: TestAccSentinelAlertRuleDataSource_basic (213.18s)
=== CONT  TestAccSentinelAutomationRule_requiresImport
--- PASS: TestAccSentinelAlertRuleNrt_requiresImport (265.70s)
=== CONT  TestAccSentinelAutomationRule_update
--- PASS: TestAccSentinelAutomationRule_requiresImport (200.21s)
=== CONT  TestAccSentinelAutomationRule_complete
--- PASS: TestAccSentinelAutomationRule_complete (168.24s)
=== CONT  TestAccSentinelAutomationRule_basic
--- PASS: TestAccSentinelAutomationRule_update (427.70s)
=== CONT  TestAccSentinelAlertRuleTemplateDataSource_nrt
--- PASS: TestAccSentinelAutomationRule_basic (224.07s)
=== CONT  TestAccSentinelAlertRuleTemplateDataSource_scheduled
--- PASS: TestAccSentinelAlertRuleTemplateDataSource_nrt (223.81s)
=== CONT  TestAccSentinelAlertRuleTemplateDataSource_securityIncident
--- PASS: TestAccSentinelAlertRuleTemplateDataSource_scheduled (211.84s)
=== CONT  TestAccSentinelAlertRuleTemplateDataSource_fusion
--- PASS: TestAccSentinelAlertRuleTemplateDataSource_securityIncident (211.63s)
=== CONT  TestAccSentinelAlertRuleScheduled_updateEventGroupingSetting
--- PASS: TestAccSentinelAlertRuleTemplateDataSource_fusion (154.67s)
=== CONT  TestAccSentinelAlertRuleScheduled_withAlertRuleTemplateGuid
--- PASS: TestAccSentinelAlertRuleScheduled_withAlertRuleTemplateGuid (158.61s)
=== CONT  TestAccSentinelAlertRuleScheduled_requiresImport
--- PASS: TestAccSentinelAlertRuleScheduled_updateEventGroupingSetting (316.42s)
=== CONT  TestAccSentinelAlertRuleScheduled_update
--- PASS: TestAccSentinelAlertRuleScheduled_requiresImport (268.79s)
=== CONT  TestAccSentinelAlertRuleScheduled_complete
--- PASS: TestAccSentinelAlertRuleScheduled_update (368.42s)
=== CONT  TestAccSentinelAlertRuleScheduled_basic
--- PASS: TestAccSentinelAlertRuleScheduled_complete (241.11s)
=== CONT  TestAccSentinelAlertRuleNrt_withAlertRuleTemplateGuid
--- PASS: TestAccSentinelAlertRuleScheduled_basic (224.24s)
=== CONT  TestAccSentinelAlertRuleMsSecurityIncident_basic
--- PASS: TestAccSentinelAlertRuleNrt_withAlertRuleTemplateGuid (236.05s)
=== CONT  TestAccSentinelAlertRuleNrt_update
--- PASS: TestAccSentinelAlertRuleMsSecurityIncident_basic (238.35s)
=== CONT  TestAccSentinelAlertRuleNrt_complete
--- PASS: TestAccSentinelAlertRuleNrt_complete (226.44s)
=== CONT  TestAccSentinelAlertRuleNrt_basic
--- PASS: TestAccSentinelAlertRuleNrt_update (448.68s)
=== CONT  TestAccSentinelAlertRuleMsSecurityIncident_withDisplayNameExcludeFilter
--- PASS: TestAccSentinelAlertRuleNrt_basic (225.27s)
=== CONT  TestAccSentinelAlertRuleMsSecurityIncident_withAlertRuleTemplateGuid
--- PASS: TestAccSentinelAlertRuleMsSecurityIncident_withAlertRuleTemplateGuid (163.56s)
=== CONT  TestAccSentinelAlertRuleMsSecurityIncident_requiresImport
--- PASS: TestAccSentinelAlertRuleMsSecurityIncident_withDisplayNameExcludeFilter (495.22s)
=== CONT  TestAccSentinelAlertRuleMsSecurityIncident_update
--- PASS: TestAccSentinelAlertRuleMsSecurityIncident_requiresImport (188.02s)
=== CONT  TestAccSentinelAlertRuleMsSecurityIncident_complete
--- PASS: TestAccSentinelAlertRuleMsSecurityIncident_complete (161.76s)
=== CONT  TestAccSentinelAlertRuleMLBehaviorAnalytics_basic
--- PASS: TestAccSentinelAlertRuleMsSecurityIncident_update (373.36s)
=== CONT  TestAccSentinelAlertRuleMLBehaviorAnalytics_requiresImport
--- PASS: TestAccSentinelAlertRuleMLBehaviorAnalytics_basic (252.50s)
=== CONT  TestAccSentinelAlertRuleMLBehaviorAnalytics_update
--- PASS: TestAccSentinelAlertRuleMLBehaviorAnalytics_requiresImport (289.39s)
=== CONT  TestAccSentinelAlertRuleMLBehaviorAnalytics_complete
--- PASS: TestAccSentinelAlertRuleMLBehaviorAnalytics_complete (235.69s)
=== CONT  TestAccSentinelAlertRuleFusion_update
--- PASS: TestAccSentinelAlertRuleMLBehaviorAnalytics_update (484.56s)
=== CONT  TestAccSentinelAlertRuleFusion_requiresImport
--- PASS: TestAccSentinelAlertRuleFusion_requiresImport (221.64s)
=== CONT  TestAccSentinelAlertRuleFusion_complete
--- PASS: TestAccSentinelAlertRuleFusion_update (395.04s)
=== CONT  TestAccSentinelAlertRuleFusion_basic
--- PASS: TestAccSentinelAlertRuleFusion_complete (234.85s)
--- PASS: TestAccSentinelAlertRuleFusion_basic (266.77s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/sentinel      4580.944s
```

</details>

### CODE REVIEW

File: internal/services/sentinel/sentinel_watchlist_resource.go [hunk 1]
```diff
@@ -131,10 +131,6 @@ func (r WatchlistResource) Create() sdk.ResourceFunc {
 					// The only supported provider for now is "Microsoft"
 					Provider: utils.String("Microsoft"),
 
-					// The "source" represent the source file name which contains the watchlist items.
```
Comment (magodo) [inline internal/services/sentinel/sentinel_watchlist_resource.go:134]: Is the `Source` removed from the new API now?

**Review by magodo — CHANGES_REQUESTED** [review]
Thanks for this PR!
I have left some additional mostly minor comments that once addressed this should be good to merge 👍

File: internal/services/sentinel/sentinel_watchlist_resource.go (hunk 1 above)
Comment (ziyeqf) [reply internal/services/sentinel/sentinel_watchlist_resource.go:134]: There does still be a `Source` property, while it seems the code about Source here only "makes API happy" without other targets. `acctests` passed without these lines, so I think it's better to remove them.

Comment (magodo) [reply internal/services/sentinel/sentinel_watchlist_resource.go:134]: Sounds reasonable.

**Review by katbyte — APPROVED** [review]
LGTM 🎩

**Comment by ziyeqf** [comment]
There seems something wrong with "NrtAlertRuleTemplate" and "2022-01-01-preview/securityinsight" after merged. please don't merge this PR till I fix them.

**Review by katbyte — APPROVED** [review]
LGTM 🚜

**Comment by ziyeqf** [comment]
issue #17910 has also been solved by this pr

**Pull request merged**